gitlab/.gitlab-ci.yml

```diff
@@ -0,0 +1,417 @@
+    - if [ ! -d /cache/sstate/$CI_COMMIT_REF_NAME ]; then mkdir /cache/sstate/${CI_COMMIT_REF_NAME}; fi
+    - ln -s /cache/sstate/${CI_COMMIT_REF_NAME} sstate-cache
+    # Download the sources for building
+    - repo init -u https://${MANIFEST_REPO} -b $MANIFEST_BRANCH --no-clone-bundle -m $MANIFEST_FILE
+    - repo sync --no-clone-bundle
+    - repo start ${CI_COMMIT_REF_NAME} openxt/*
+    # Configure the build
+    - openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} config -t stable-9 -b ${CI_COMMIT_REF_NAME}
+    # Copy the certs into the workspace
+    - mkdir -p certs && cp -R /certs/* certs/
+    # Update the mirrors
+    - cd build-${CI_COMMIT_REF_NAME}/conf
+    - awk '1;/MIRRORS .*" \\/{print "    http://v3.sk/~lkundrak/dev86/archive/             https://openxt.ainfosec.com/mirror/ \\n \\"}' local.conf > local.conf.new
+    - mv local.conf local.conf.orig
+    - mv local.conf.new local.conf
+    - awk '1;/MIRRORS .*" \\/{print "    https://cgit.freedesktop.org/~airlied/vbetool/snapshot/             https://openxt.ainfosec.com/mirror/ \\n \\"}' local.conf > local.conf.new
+    - mv local.conf local.conf.orig
+    - mv local.conf.new local.conf
+    - awk '1;/MIRRORS .*" \\/{print "    http://www.ivmaisoft.com/_bin/atomic_ops/             https://ftp.osuosl.org/pub/blfs/conglomeration/libatomic/ \\n \\"}' local.conf > local.conf.new
+    - mv local.conf local.conf.orig
+    - mv local.conf.new local.conf
+  artifacts:
+    paths:
+      - openxt-workspace/
+    when: always
+    expire_in: 3 days
+  only:
+    variables:
+      - $BUILD_TYPE != ""
+
+.build_template: &build_definition  # Template for building images
+  <<: *pull-push-cache
+  image: ${EC2_DOCKER_REGISTRY}/build_image:debian_jessie
+  stage: build
+  before_script: *setup_cm
+  script:
+    - cd openxt-workspace
+    - DEPLOY_DIR=build-${CI_COMMIT_REF_NAME}/deploy
+    # Update the build/deploy directory with any cached contents, excluding IPK's
+    - if [ -d ${DEPLOY_DIR} ]; then rm -rf ${DEPLOY_DIR}/*; else mkdir -p ${DEPLOY_DIR}; fi
+    - mv ../deploy/* ${DEPLOY_DIR} || true
+    - rm -rf ${DEPLOY_DIR}/ipk || true
+    - if [[ "x$DEBUG" != "x" ]]; then ls -al build-${CI_COMMIT_REF_NAME}; ls -al ..; fi
+    # Update all the sources we're going to build, then build it
+    - repo sync --no-clone-bundle
+    - echo "openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} build \"$IMAGE_TO_BUILD\""
+    - openxt/bordel/bordel -i ${CI_COMMIT_REF_NAME} build "$IMAGE_TO_BUILD"
+  after_script:
+    - BUILD_DIR=openxt-workspace/build-${CI_COMMIT_REF_NAME}
+    # Ensure we have the log and deploy directories for caching
+    - mkdir -p {deploy,logs}
+    # Save the cooker logs, then get rid of the tmp* directories, they're too large
+    # Modify this if you need to debug your compile
+    - find ${BUILD_DIR}/tmp-glibc/log/cooker -name "20*" -exec cp {} logs \;
+    # Get rid of the tmp* directories, they're too large
+    - find ${BUILD_DIR}/tmp-glibc -type f -delete || true
```

Comment:
When doing a full build of all images, this was considerably faster for me.

Reply:
That makes me think something is afoot here (especially with `|| true`).

It looks like the intent is to remove build artifacts that would have been created, but are no longer useful, in the tmp-dir for each recipe. That should be covered by `rm-work.bbclass`, which would take advantage of OE's parallelism. You can enable it with `bordel config --rmwork` or by doing what it does at more appropriate time.

As far as I know `find[...] -delete` does not run deletion in parallel, so I cannot quite see why it would be faster than `rm`.

Reply:
I think I may have been using `rm -rf ${BUILD_DIR}/tmp-glibc/*` which takes a long time to find all the files first and then delete them. It appears that removing the `/*` should fix the speed issue.